src/neptune_scale/sync/metadata_splitter.py

```diff
@@ -251,3 +304,40 @@ def _make_preview(self) -> Optional[Preview]:
         if self._metrics_preview_completion is not None:
             return Preview(is_preview=True, completion_ratio=self._metrics_preview_completion)
         return Preview(is_preview=True)
+
+
+def datetime_to_proto(dt: datetime) -> Timestamp:
+    dt_ts = dt.timestamp()
+    return Timestamp(seconds=int(dt_ts), nanos=int((dt_ts % 1) * 1e9))
+
+
+def make_step(number: Union[float, int], raise_on_step_precision_loss: bool = False) -> Step:
+    """
+    Converts a number to protobuf Step value. Example:
+    >>> assert make_step(7.654321, True) == Step(whole=7, micro=654321)
+
+    Args:
+        number: step expressed as number
+        raise_on_step_precision_loss: inform converter whether it should silently drop precision and
+            round down to 6 decimal places or raise an error.
+
+    Returns: Step protobuf used in Neptune API.
+    """
+    m = int(1e6)
+    micro: int = int(number * m)
+    if raise_on_step_precision_loss and number * m - micro != 0:
+        raise ValueError(f"step must not use more than 6-decimal points, got: {number}")
+
+    whole = micro // m
+    micro = micro % m
+
+    return Step(whole=whole, micro=micro)
```

Comment:
This version is ~8% faster: immediate value instead of local var lookup + one substraction less:
```suggestion
    micro: int = int(number * 1_000_000)
    if raise_on_step_precision_loss and number * 1_000_000 != micro:
        raise ValueError(f"step must not use more than 6-decimal points, got: {number}")

    whole = micro // 1_000_000
    micro = micro % 1_000_000

    return Step(whole=whole, micro=micro)
```